src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java

```diff
@@ -119,25 +122,34 @@ public String getApiUri() {
 
     @Override public void visitSources(SCMSourceObserver observer) throws IOException, InterruptedException {
         TaskListener listener = observer.getListener();
-        if (repoOwner.isEmpty()) {
-            listener.getLogger().format("Must specify user or organization%n");
-            return;
-        }
         StandardCredentials credentials = Connector.lookupScanCredentials(observer.getContext(), apiUri, scanCredentialsId);
+
+        // Github client and validation
         GitHub github = Connector.connect(apiUri, credentials);
+        try {
+            github.checkApiUrlValidity();
```

Comment:
You are missing my point. There are two cases:

* By far the most common: GitHub is accessible, and any errors that might occur during scanning are about real problems. This PR adds another network round-trip, slowing indexing and adding a bit to your rate limit.
* Occasionally, GitHub is inaccessible. This PR reports the problem nicely. But it could also be reported by going ahead and trying normal operations to get real information and specially rendering exceptions which we know are generally caused by network problems.